File: weyl-core/src/main/java/io/quartic/weyl/core/alert/Alert.java
package io.quartic.weyl.core.alert;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import org.immutables.value.Value;

import java.util.Optional;

import static io.quartic.weyl.core.alert.Alert.Level.INFO;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = AlertImpl.class)
@JsonDeserialize(as = AlertImpl.class)
public interface Alert {
    enum Level {
        INFO,
        WARNING,
        SEVERE
    }

    String title();
    Optional<String> body();
    @Value.Default
    default Level level() {
        return INFO;
    }
}

File: weyl-core/src/main/kotlin/io/quartic/weyl/core/model/Alert.kt
package io.quartic.weyl.core.model

import io.quartic.weyl.core.model.Alert.Level.INFO

data class Alert(
        val title: String,
        val body: String?,
        val level: Level = INFO
) {
    enum class Level {
        INFO,
        WARNING,
        SEVERE
    }
}
